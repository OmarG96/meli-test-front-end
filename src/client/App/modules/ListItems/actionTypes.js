export const SEARCH_FETCH = "SEARCH_FETCH";
export const SEARCH_FETCH_SUCCESS = "SEARCH_FETCH_SUCCESS";
export const SEARCH_FETCH_ERROR = "SEARCH_FETCH_ERROR";
