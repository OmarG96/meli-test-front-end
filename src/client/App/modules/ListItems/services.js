import { http } from "../../../utils/http";

export const searchItems = async (item) => {
  const response = await http.get("/api/items", {
    params: {
      q: item,
    },
  });

  return response.data;
};
