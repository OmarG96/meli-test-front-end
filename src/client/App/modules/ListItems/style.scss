.breadcrumb-container {
  grid-column: 2 / span 10;
  height: 48px;
  width: 100%;
}
