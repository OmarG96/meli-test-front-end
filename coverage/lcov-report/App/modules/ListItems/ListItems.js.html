<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for App/modules/ListItems/ListItems.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">App/modules/ListItems</a> ListItems.js
    </h1>
    <div class='clearfix'>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
&nbsp;
import { useQueryURL } from "../../../utils/hooks";
import { onFetch } from "./actions";
&nbsp;
import { CardContainer, GridContainer, Breadcrumb } from "../../../common";
import { Item } from "./components";
&nbsp;
import "./style.scss";
&nbsp;
const ListItems = () =&gt; {
  const dispatch = useDispatch();
  const query = useQueryURL();
  const itemToSearch = query.get("search");
&nbsp;
  const { data, loading, error } = useSelector(
    ({ listItems: { search } }) =&gt; search
  );
  const { items = [], categories = [] } = data || {};
&nbsp;
  useEffect(() =&gt; {
    dispatch(onFetch(itemToSearch));
  }, [itemToSearch, dispatch]);
&nbsp;
  return (
    &lt;GridContainer&gt;
      &lt;Breadcrumb path={categories} /&gt;
      {items.length ? (
        &lt;CardContainer&gt;
          {items.map((itemData) =&gt; (
            &lt;Item key={itemData.id} data={itemData} /&gt;
          ))}
        &lt;/CardContainer&gt;
      ) : null}
    &lt;/GridContainer&gt;
  );
};
&nbsp;
export default ListItems;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jun 21 2021 18:32:13 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
