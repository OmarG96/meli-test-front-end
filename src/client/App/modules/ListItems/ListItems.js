import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";

import { useQueryURL } from "../../../utils/hooks";
import { onFetch } from "./actions";

import { CardContainer, GridContainer, Breadcrumb } from "../../../common";
import { Item } from "./components";

import "./style.scss";

const ListItems = () => {
  const dispatch = useDispatch();
  const query = useQueryURL();
  const itemToSearch = query.get("search");

  const { data, loading, error } = useSelector(
    ({ listItems: { search } }) => search
  );
  const { items = [], categories = [] } = data || {};

  useEffect(() => {
    dispatch(onFetch(itemToSearch));
  }, [itemToSearch, dispatch]);

  return (
    <GridContainer>
      <Breadcrumb path={categories} />
      {items.length ? (
        <CardContainer>
          {items.map((itemData) => (
            <Item key={itemData.id} data={itemData} />
          ))}
        </CardContainer>
      ) : null}
    </GridContainer>
  );
};

export default ListItems;
