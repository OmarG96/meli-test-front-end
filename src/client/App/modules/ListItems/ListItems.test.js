import React from "react";
import renderWithProvider from "../../../utils/tests/renderWithProvider";
import ListItems from ".";
import { screen } from "@testing-library/react";
import { SEARCH_FETCH, SEARCH_FETCH_SUCCESS } from "../ListItems/actionTypes";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "http://localhost:3000/items?search=EXAMPLE123",
  }),
}));

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => jest.fn(),
}));

jest.mock("./actions", () => ({
  ...jest.requireActual("./actions"),
  onFetch: jest.fn(),
}));

describe("GIVEN ListItems", () => {
  describe("WHEN is rendered", () => {
    it("THEN match to snapshot", async () => {
      const { asFragment } = renderWithProvider(<ListItems />, {
        onPrepareStore: (store) =>
          store.dispatch({
            type: SEARCH_FETCH_SUCCESS,
            payload: {
              data: {
                categories: ["example1", "example2", "example3"],
                items: [
                  {
                    id: "EXAMPLE123",
                    title: "Title title title",
                    price: {
                      currency: "ARS",
                      amount: "123456",
                    },
                    picture: "url.example.com",
                    condition: "new",
                    free_shipping: true,
                    sold_quantity: 123,
                    description:
                      "description description description. Description.",
                  },
                ],
              },
            },
          }),
      });

      await screen.findByText("Title title title");
      expect(asFragment()).toMatchSnapshot();
    });
    it("THEN match to snapshot when items is empty", async () => {
      const { asFragment } = renderWithProvider(<ListItems />, {
        onPrepareStore: (store) =>
          store.dispatch({
            type: SEARCH_FETCH,
          }),
      });

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
