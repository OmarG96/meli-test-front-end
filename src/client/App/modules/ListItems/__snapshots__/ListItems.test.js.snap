// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN ListItems WHEN is rendered THEN match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid-container-root"
  >
    <div
      class="breadcrumb-root"
    >
      <span
        class="breadcrumb-route"
      >
        example1
      </span>
      <span
        class="breadcrumb-route"
      >
        example2
      </span>
      <span
        class="breadcrumb-route"
      >
        example3
      </span>
    </div>
    <div
      class="card-container-root"
    >
      <div
        class="item-root"
        role="item"
      >
        <div
          class="item"
        >
          <div
            class="picture"
          >
            <img
              src="url.example.com"
            />
          </div>
          <div
            class="detail"
          >
            <div
              class="header"
            >
              <div
                class="price"
              >
                $ 123,456
                <img
                  class="icon-shipping"
                  src="test-file-stub"
                />
              </div>
              <div
                class="state-name"
              />
            </div>
            <div
              class="title"
            >
              Title title title
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GIVEN ListItems WHEN is rendered THEN match to snapshot when items is empty 1`] = `
<DocumentFragment>
  <div
    class="grid-container-root"
  >
    <div
      class="breadcrumb-root"
    />
  </div>
</DocumentFragment>
`;
